spotinst/elastigroup_gcp_strategy/fields_spotinst_elastigroup_gcp_strategy.go

```diff
@@ -230,4 +231,139 @@ func Setup(fieldsMap map[commons.FieldName]*commons.GenericField) {
+		},
+		func(resourceObject interface{}, resourceData *schema.ResourceData, meta interface{}) error {
+			egWrapper := resourceObject.(*commons.ElastigroupGCPWrapper)
+			elastigroup := egWrapper.GetElastigroup()
+			if v, ok := resourceData.GetOk(string(OptimizationWindow)); ok {
+				optimizationWindowList := v.([]interface{})
+				optimizationWindow := make([]string, len(optimizationWindowList))
+				for i, j := range optimizationWindowList {
+					optimizationWindow[i] = j.(string)
+				}
+				elastigroup.Strategy.SetOptimizationWindows(optimizationWindow)
+			} else {
+				elastigroup.Strategy.SetOptimizationWindows(nil)
+			}
+			return nil
+		},
+		nil,
+	)
+	fieldsMap[RevertToPreemptible] = commons.NewGenericField(
+		commons.ElastigroupGCPStrategy,
+		RevertToPreemptible,
+		&schema.Schema{
+			Type:     schema.TypeList,
+			Optional: true,
+			Elem: &schema.Resource{
+				Schema: map[string]*schema.Schema{
+					string(PerformAt): {
+						Type:     schema.TypeString,
+						Required: true,
+					},
+				},
+			},
+		},
+		func(resourceObject interface{}, resourceData *schema.ResourceData, meta interface{}) error {
+			egWrapper := resourceObject.(*commons.ElastigroupGCPWrapper)
+			elastigroup := egWrapper.GetElastigroup()
+			var result interface{} = nil
+			if elastigroup.Strategy != nil && elastigroup.Strategy.RevertToPreemptible != nil &&
+				elastigroup.Strategy.RevertToPreemptible.PerformAt != nil {
+				revertToPreemptible := elastigroup.Strategy.RevertToPreemptible
+				result = flattenRevertToPreemptible(revertToPreemptible)
+			}
+			if result != nil {
+				if err := resourceData.Set(string(RevertToPreemptible), result); err != nil {
+					return fmt.Errorf(string(commons.FailureFieldReadPattern), string(RevertToPreemptible), err)
+				}
+			}
+			return nil
+		},
+		func(resourceObject interface{}, resourceData *schema.ResourceData, meta interface{}) error {
+			egWrapper := resourceObject.(*commons.ElastigroupGCPWrapper)
+			elastigroup := egWrapper.GetElastigroup()
+			if v, ok := resourceData.GetOk(string(RevertToPreemptible)); ok {
+				if revertToPreemptible, err := expandRevertToPreemptible(v); err != nil {
+					return err
+				} else {
+					elastigroup.Strategy.SetRevertToPreemptible(revertToPreemptible)
+				}
+			}
+			return nil
+		},
+		func(resourceObject interface{}, resourceData *schema.ResourceData, meta interface{}) error {
+			egWrapper := resourceObject.(*commons.ElastigroupGCPWrapper)
+			elastigroup := egWrapper.GetElastigroup()
+			var result *gcp.RevertToPreemptible = nil
+			if v, ok := resourceData.GetOk(string(RevertToPreemptible)); ok {
+				if value, err := expandRevertToPreemptible(v); err != nil {
+					return err
+				} else {
+					result = value
+				}
+			}
+			elastigroup.Strategy.SetRevertToPreemptible(result)
+			return nil
+		},
+		nil,
+	)
+}
+func flattenRevertToPreemptible(revertToPreemptible *gcp.RevertToPreemptible) []interface{} {
+	result := make([]interface{}, 0, 1)
```

Comment:
why hardcoded to 0, 1

Reply:
Because revertToPreemptible is an pointer to object and not an array so we cant use len() to create list of interface{}.